tests/distributed/test_multi_node_topology.py

```diff
@@ -0,0 +1,70 @@
+"""Compare the outputs of HF and distributed vLLM when using greedy sampling.
+
+Run:
+```sh
+cd $VLLM_PATH/tests
+
+pytest distributed/test_multi_node.py
+```
+"""
+
+import pytest
+import ray
+from ray.cluster_utils import Cluster
+
+from vllm.utils import cuda_device_count_stateless
+
+
+@pytest.mark.skipif(cuda_device_count_stateless() < 2,
+                    reason="Need at least 2 GPUs to run the test.")
+@pytest.mark.parametrize("model, distributed_executor_backend, test_suite", [
+    ("facebook/opt-125m", "ray"),
+])
+def test_multi_node_bad_topology(
```

Comment:
I think you should test it with 2 nodes test (where ray cluster is already created), and try to launch the test with 1 gpus.

Reply:
hmm yeah I think we need real 2 nodes if we want to use node ip resources. let me fix it.